### PR TITLE
Feat/civ 2691 civ 2898 evidence upload

### PR DESCRIPTION
**Pull request by bhagyashreesharma90**
### Change description ###
https://tools.hmcts.net/jira/browse/CIV-2691

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandler.java [hunk 1]
```diff
@@ -4,20 +4,43 @@
...
 
     @EventListener
     public void sendEvidenceUploadNotification(EvidenceUploadNotificationEvent event) {
-        coreCaseDataService.triggerEvent(event.getCaseId(), EVIDENCE_UPLOAD_NOTIFICATION);
+        CaseDetails caseDetails = coreCaseDataService.getCase(event.getCaseId());
+        CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
+        try {
+            evidenceUploadApplicantNotificationHandler.notifyApplicantEvidenceUpload(caseData);
+        } catch (Exception e) {
+            log.error("Failed to send email notification to applicant for case '{}'", event.getCaseId());
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandler.java:33]: Add a comment in this method or in each of the "catch" clauses explaining why it is acceptable that we log but don't rethrow. Something like "these email notifications are not critical to the business and provided as a courtesy, as the user can log in and see the new uploads regardless. Hence we treat any failure here just as a warning"

Also: if it's not critical that emails are sent, let's not pollute the "errors" log. Use `log.warn` instead.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandlerTest.java [hunk 2]
```diff
@@ -5,26 +5,82 @@
...
+            .build();
+        CaseDetails caseDetails = CaseDetailsBuilder.builder().data(caseData).build();
+        when(coreCaseDataService.getCase(caseId)).thenReturn(caseDetails);
+        when(caseDetailsConverter.toCaseData(caseDetails)).thenReturn(caseData);
+        EvidenceUploadNotificationEvent event = new EvidenceUploadNotificationEvent(caseId);
+        handler.sendEvidenceUploadNotification(event);
+
+        verify(applicantNotificationHandler).notifyApplicantEvidenceUpload(caseData);
+        verify(respondentNotificationHandler).notifyRespondentEvidenceUpload(caseData, false);
+        verify(respondentNotificationHandler).notifyRespondentEvidenceUpload(caseData, true);
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandlerTest.java:50]: Add `//Given`, `//When`, `//Then` sections to the tests for clarity.

**Review by filippo-possenti-version1 — CHANGES_REQUESTED** [review]
A couple of small changes and we're good to go.

File: src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadApplicantNotificationHandlerTest.java [hunk 3]
```diff
@@ -53,11 +45,7 @@ void setup() {
         void shouldNotifyApplicantSolicitor_whenInvoked() {
             //given: case where applicant solicitor has email as applicantsolicitor@example.com
             CaseData caseData = CaseDataBuilder.builder().atStateClaimDetailsNotified().build();
-            CallbackParams params = CallbackParamsBuilder.builder().of(ABOUT_TO_SUBMIT, caseData).build();
-
-            //when: about to submit is called for callback handler
-            handler.handle(params);
-
+            handler.notifyApplicantEvidenceUpload(caseData);
```
Comment (hmcts-version1-ignacio) [inline src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadApplicantNotificationHandlerTest.java:48]: //when: Evidence for Applicant is uploaded

File: src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadRespondentNotificationHandlerTest.java [hunk 4]
```diff
@@ -56,13 +46,7 @@ void setup() {
...
             //given: case data has one respondent solicitor
             CaseData caseData = CaseDataBuilder.builder().atStateClaimDetailsNotified().build();
-            CallbackParams params = CallbackParamsBuilder.builder().of(ABOUT_TO_SUBMIT, caseData)
-                .request(CallbackRequest.builder().eventId(NOTIFY_RESPONDENT_SOLICITOR1_FOR_EVIDENCE_UPLOAD.name())
-                             .build()).build();
-
-            //when: about to submit is called for callback handler with respondent1 event
-            handler.handle(params);
-
+            handler.notifyRespondentEvidenceUpload(caseData, true);
```
Comment (hmcts-version1-ignacio) [inline src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadRespondentNotificationHandlerTest.java:49]: //when: Evidence for Respondent is uploaded

File: src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadRespondentNotificationHandlerTest.java [hunk 5]
```diff
@@ -79,13 +63,7 @@ void shouldNotifyRespondent2Solicitor_whenInvoked() {
...
                 .respondentSolicitor2EmailAddress("respondentsolicitor@example.com")
                 .build();
-            CallbackParams params = CallbackParamsBuilder.builder().of(ABOUT_TO_SUBMIT, caseData)
-                .request(CallbackRequest.builder()
-                             .eventId(NOTIFY_RESPONDENT_SOLICITOR2_FOR_EVIDENCE_UPLOAD.name()).build()).build();
-
-            //when: about to submit is called for callback handler with respondent2 event
-            handler.handle(params);
-
+            handler.notifyRespondentEvidenceUpload(caseData, false);
```
Comment (hmcts-version1-ignacio) [inline src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadRespondentNotificationHandlerTest.java:66]: //when: Evidence for Respondent Solicitor 2 is uploaded

File: src/test/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandlerTest.java (hunk 2 above)
Comment (bhagyashreesharma90) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandlerTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadRespondentNotificationHandlerTest.java (hunk 5 above)
Comment (bhagyashreesharma90) [reply src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadRespondentNotificationHandlerTest.java:68]: done

File: src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadApplicantNotificationHandlerTest.java (hunk 3 above)
Comment (bhagyashreesharma90) [reply src/test/java/uk/gov/hmcts/reform/civil/notification/EvidenceUploadApplicantNotificationHandlerTest.java:49]: done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandler.java (hunk 1 above)
Comment (bhagyashreesharma90) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandler.java]: done

**Pull request merged**